lightning/src/ln/channel.rs

```diff
@@ -4844,20 +4844,25 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<&'a K> for Channel<Signer>
 		let counterparty_dust_limit_satoshis = Readable::read(reader)?;
 		let holder_dust_limit_satoshis = Readable::read(reader)?;
 		let counterparty_max_htlc_value_in_flight_msat = Readable::read(reader)?;
-		let mut counterparty_selected_channel_reserve_satoshis = Some(Readable::read(reader)?);
-		if counterparty_selected_channel_reserve_satoshis == Some(0) {
-			// Versions up to 0.0.98 had counterparty_selected_channel_reserve_satoshis as a
-			// non-option, writing 0 for what we now consider None.
-			counterparty_selected_channel_reserve_satoshis = None;
+		let mut counterparty_selected_channel_reserve_satoshis = None;
+		if ver == 1 {
+			// Read the old serialization from version 0.0.98.
+			counterparty_selected_channel_reserve_satoshis = Some(Readable::read(reader)?);
+		} else {
+			// Read the 8 bytes of backwards-compatibility data.
+			let _dummy: u64 = Readable::read(reader)?;
 		}
 		let counterparty_htlc_minimum_msat = Readable::read(reader)?;
 		let holder_htlc_minimum_msat = Readable::read(reader)?;
 		let counterparty_max_accepted_htlcs = Readable::read(reader)?;
-		let mut minimum_depth = Some(Readable::read(reader)?);
-		if minimum_depth == Some(0) {
-			// Versions up to 0.0.98 had minimum_depth as a non-option, writing 0 for what we now
-			// consider None.
-			minimum_depth = None;
+
+		let mut minimum_depth = None;
+		if ver == 1 {
+			// Read the old serialization from version 0.0.98.
+			minimum_depth = Some(Readable::read(reader)?);
+		} else {
+			// Read the 4 bytes of backwards-compatibility data.
+			let _dummy: u32 = Readable::read(reader)?;
```

Comment:
Could be nice to have the corresponding comment in the `Writer`, at `minimum_depth`/`counterparty_selected_channel_reserve_satoshis`, "write the field for backwards-compatibility data" ?